_posts/2020-06-19-async-explorer.markdown

```diff
@@ -0,0 +1,99 @@
+---
+layout: post
+title:  "Async Explorer"
+date:   2020-06-19 16:24:08 +0300
+categories: jekyll update
+tags: async
+---
+
+Have you ever looked up into the stars and wondered, "How the fuck is that feature implemented"? In this series, I'll (hopefully) dive into the implementation of coroutines for several (compiled) programming languages.
+
+A short disclaimer: I'm not too sharp on the details of some (or actually any) of these implementations. Most of this will just be me rambling and looking at compiler source code/the output of [godbolt](https://godbolt.org/). I try really hard to validate every claim I'll post here, but some mistakes are sure to sneak their way into one of these. Feel free to point them up and I'll fix them as soon as I can.
+
+## Purpose
+
+The purpose of this post is to set a few definitions for my upcoming posts. I'll explain what I mean when I use words like "coroutine" or "asynchronous", which will hopefully allow me to use them freely in the following posts.
+
+Afterwards, I'll give a few short examples of how to coroutines "look like" in several programming languages, like C++, Rust, Zig and the LLVM IR. 
+
+I hope I'll also be able to give you a rough idea of what coroutines are usful for, but it's not the main goal of this post.
+
+This is not an "asyncio" tutorial or anything like that, just me spewing nonsense and messing with some programming languages.
+
+## What is async?
+
+If you're like me, you probably heard terms like "async" and "coroutines" mentioned a lot, and maybe even saw (or wrote) some asynchronious code, but didn't really understand what's going on behind the scenes. Atleast I didn't (and still don't).
```

Comment:
You can add a small list of well known asyc libs/modules. It will connect connect the reader to the term

Reply:
I've added that under examples